tests/sklearn/test_sklearn_autolog.py

```diff
@@ -1462,3 +1462,275 @@ def test_metric_computation_handles_absent_labels():
+                as patched_log_post_training_metric:
+            # test post training metric logging disabled in fit scope
+            lr_model.fit(X, y)
+            patched_log_post_training_metric.assert_not_called()
+
+            patched_log_post_training_metric.reset_mock()
+            # test post training metric logging called only once in model.score
+            lr_model.score(eval1_X, eval1_y)
+            assert patched_log_post_training_metric.call_count == 1
+            assert patched_log_post_training_metric.call_args[0][1] == 'LinearRegression_score_eval1_X'
+
+            patched_log_post_training_metric.reset_mock()
+            # test post training metric logging disabled in eval_and_log_metrics
+            mlflow.sklearn.eval_and_log_metrics(lr_model, eval1_X, eval1_y, prefix='test1')
+            patched_log_post_training_metric.assert_not_called()
+
+
+def test_multi_model_interleaved_fit_and_post_train_metric_call():
+    import sklearn.linear_model
+    mlflow.sklearn.autolog()
+    from sklearn.metrics import mean_squared_error
+
+    X, y = get_iris()
+    eval1_X, eval1_y = X[0::3], y[0::3]
+    eval2_X, eval2_y = X[1::3], y[1::3]
+
+    lr_model1 = sklearn.linear_model.LinearRegression(fit_intercept=True)
+    lr_model2 = sklearn.linear_model.LinearRegression(fit_intercept=False)
+
+    with mlflow.start_run() as run1:
+        lr_model1.fit(X, y)
+
+    with mlflow.start_run() as run2:
+        lr_model2.fit(X, y)
+
+    model1_r2_score = lr_model1.score(eval1_X, eval1_y)
+    model2_r2_score = lr_model2.score(eval2_X, eval2_y)
+
+    pred1_y = lr_model1.predict(eval1_X)
+    model1_mse = mean_squared_error(eval1_y, pred1_y)
+
+    pred2_y = lr_model2.predict(eval2_X)
+    model2_mse = mean_squared_error(eval2_y, pred2_y)
+
+    _, metrics1, _, _ = get_run_data(run1.info.run_id)
+    assert metrics1['LinearRegression_score_eval1_X'] == model1_r2_score
+    assert metrics1['mean_squared_error_eval1_X'] == model1_mse
+
+    _, metrics2, _, _ = get_run_data(run2.info.run_id)
+    assert metrics2['LinearRegression_score_eval2_X'] == model2_r2_score
+    assert metrics2['mean_squared_error_eval2_X'] == model2_mse
+
+
+@pytest.mark.parametrize('scoring', [None, sklearn.metrics.make_scorer(sklearn.metrics.accuracy_score)])
+def test_meta_estimator_disable_post_training_autologging(scoring):
+    import sklearn.svm
+    import sklearn.metrics
+    mlflow.sklearn.autolog()
+
+    X, y = get_iris()
+    with mock.patch("mlflow.sklearn._AutologgingMetricsManager.register_model") as mock_register_model, \
+            mock.patch("mlflow.sklearn._AutologgingMetricsManager.is_metric_value_loggable") as mock_is_metric_value_loggable, \
+            mock.patch("mlflow.sklearn._AutologgingMetricsManager.log_post_training_metric") as mock_log_post_training_metric, \
+            mock.patch("mlflow.sklearn._AutologgingMetricsManager.register_prediction_input_dataset") as mock_register_prediction_input_dataset:
```

Comment:
@WeichenXu123 
It's not necessary to apply this change, but `mock.patch.multiple` can be useful when patching multiple methods:

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.multiple